### PR TITLE
FIX-#2269: Move `default_to_pandas` logic from API layer to backend

### LINKED ISSUE #2269: Move defaulting to pandas at groupby aggregation from API level to backend
Hi! I have been working on some other backend for modin, and I have seen that modin defaults multiple column groupby to pandas. I have two questions/concerns:

1. Why if I pass a one element list ['column_title'], modin classifies this as a _is_multi_by? In particular, I am wondering what is the logic behind these lines of code in groupby.py
```
self._is_multi_by = (
                isinstance(by, type(self._query_compiler)) and len(by.columns) > 1
            ) or (
                not isinstance(by, type(self._query_compiler))
                and axis == 0
                and all(obj in self._query_compiler.columns for obj in self._by)
            )
```
the first condition makes sense to me, but the second one is throwing me off.

2. Why is the reason behind defaulting multi_col index to pandas? I would feel that the engine should take care of this.

Thanks!
LP

### PR DESCRIPTION
**Pull request by YarShev**
Signed-off-by: Igoshev, Yaroslav <yaroslav.igoshev@intel.com>

## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/developer/contributing.html)
- [x] passes `flake8 modin`
- [x] passes `black --check modin`
- [x] signed commit with `git commit -s` 
- [x] Resolves #2269  
- [x] tests added and passing

### CODE REVIEW

File: modin/backends/pandas/query_compiler.py [hunk 1]
```diff
@@ -2583,24 +2630,54 @@ def groupby_dict_agg(self, by, func_dict, groupby_args, agg_args, drop=False):
...
+        groupby_kwargs,
+        drop_,
+        drop=False,
+    ):
+        agg_func = wrap_udf_function(agg_func)
+
+        if is_multi_by:
+            return self._default_to_pandas_groupby(
+                agg_func, by, axis, drop, groupby_kwargs, *agg_args, **agg_kwargs
+            )
```
Comment (dchigarev) [inline modin/backends/pandas/query_compiler.py:2651]: to be consistent we can replace it to
```python
return super().groupby_agg(...)
```
I've run tests locally with this change and they passed

File: modin/pandas/groupby.py [hunk 2]
```diff
@@ -834,28 +834,16 @@ def _apply_agg_function(self, f, drop=True, *args, **kwargs):
...
-        else:
-            by = self._by
-
-        # For aggregations, pandas behavior does this for the result.
-        # For other operations it does not, so we wait until there is an aggregation to
-        # actually perform this operation.
-        if self._idx_name is not None and drop and self._drop:
-            groupby_qc = self._query_compiler.drop(columns=[self._idx_name])
-        else:
-            groupby_qc = self._query_compiler
```
Comment (dchigarev) [inline modin/pandas/groupby.py:852]: why this logic was moved to backend?

File: modin/backends/pandas/query_compiler.py (hunk 1 above)
Comment (YarShev) [reply modin/backends/pandas/query_compiler.py:2651]: Looks like it makes sense.

File: modin/pandas/groupby.py (hunk 2 above)
Comment (YarShev) [reply modin/pandas/groupby.py:852]: Because it depends on `is_multi_by` which is checked above.

Comment (dchigarev) [reply modin/pandas/groupby.py:852]: well, then we can just add `if not self._is_multi_by and ...` to that if statement?

Comment (dchigarev) [reply modin/pandas/groupby.py:852]: I'll try to move that logic back, and post about results (is it possible or not)

Comment (YarShev) [reply modin/pandas/groupby.py]: Feel free to push these changes into this branch if everything is ok.

Comment (dchigarev) [reply modin/pandas/groupby.py]: done it

**Review by gshimansky — APPROVED** [review]

File: modin/backends/pandas/query_compiler.py (hunk 1 above)
Comment (gshimansky) [reply modin/backends/pandas/query_compiler.py]: This change broke "default to pandas" warning message. Now we have strange message `UserWarning: <lambda> defaulting to pandas implementation.` that doesn't explain anything.

**Pull request merged**